Count default test compares a string with an integer

Seeing this failure:

"java.lang.AssertionError: A value of 10 for constraint CountDefault is expected expected <br><br>[10] but found <br><br>[10]"

I've checked the code, it seems it's comparing a string with an integer, and thus always failing:

https://github.com/opengeospatial/ets-wfs20-nsg/blob/c3885eeaa04adce6cef0958a8b22951546086526/src/main/java/org/opengis/cite/wfs20/nsg/testsuite/getfeature/CountParameter.java#L47

--- a/src/main/java/org/opengis/cite/wfs20/nsg/testsuite/getfeature/CountParameter.java
+++ b/src/main/java/org/opengis/cite/wfs20/nsg/testsuite/getfeature/CountParameter.java
@@ -9,6 +9,7 @@ import static org.opengis.cite.iso19142.util.ServiceMetadataUtils.getConstraintV
 import static org.opengis.cite.iso19142.util.WFSMessage.appendSimpleQuery;
 import static org.opengis.cite.wfs20.nsg.utils.NamespaceUtils.withStandardBindings;
 import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
 
 import java.util.Map;
 import java.util.Set;
@@ -44,9 +45,9 @@ public class CountParameter extends QueryFilterFixture {
     public void countDefaultIs10( ITestContext testContext ) {
         this.wfsMetadata = (Document) testContext.getSuite().getAttribute( TEST_SUBJECT.getName() );
         String constraintValueCountDefault = getConstraintValue( this.wfsMetadata, "CountDefault" );
-        assertEquals( constraintValueCountDefault, COUNT_DEFAULT_VALUE,
+        assertNotNull( constraintValueCountDefault, "A constraint CountDefault is expected" );
+        assertEquals( constraintValueCountDefault, Integer.toString( COUNT_DEFAULT_VALUE ),
                       "A value of 10 for constraint CountDefault is expected" );
-
     }
 
     @Test(description = "See NSG WFS 2.0 Profile: Requirement 13 + 14")

--- a/src/test/java/org/opengis/cite/wfs20/nsg/testsuite/getfeature/CountParameterTest.java
+++ b/src/test/java/org/opengis/cite/wfs20/nsg/testsuite/getfeature/CountParameterTest.java
@@ -1,0 +1,40 @@
+package org.opengis.cite.wfs20.nsg.testsuite.getfeature;
+
+import static org.opengis.cite.wfs20.nsg.TestUtils.createTestContext;
+
+import org.junit.Test;
+import org.testng.ITestContext;
+
+/**
+ * @author <a href="mailto:goltz@lat-lon.de">Lyn Goltz </a>
+ */
+public class CountParameterTest {
+
+    @Test
+    public void testCountParameter()
+                            throws Exception {
+        ITestContext testContext = createTestContext( "/capabilities-simple2.xml" );
+
+        CountParameter countParameter = new CountParameter();
+        countParameter.countDefaultIs10( testContext );
+    }
+
+    @Test(expected = AssertionError.class)
+    public void testCountParameter_Missing()
+                            throws Exception {
+        ITestContext testContext = createTestContext( "/capabilities-simple.xml" );
+
+        CountParameter countParameter = new CountParameter();
+        countParameter.countDefaultIs10( testContext );
+    }
+
+    @Test(expected = AssertionError.class)
+    public void testCountParameter_Unexpected()
+                            throws Exception {
+        ITestContext testContext = createTestContext( "/capabilities-withFailures.xml" );
+
+        CountParameter countParameter = new CountParameter();
+        countParameter.countDefaultIs10( testContext );
+    }
+
+}

--- a/src/test/resources/capabilities-withFailures.xml
+++ b/src/test/resources/capabilities-withFailures.xml
@@ -146,6 +146,9 @@
       <ValuesReference ows:reference="http://www.w3.org/2001/XMLSchema#boolean"/>
       <DefaultValue>FALSE</DefaultValue>
     </Constraint>
+    <Constraint name="CountDefault">
+      <DefaultValue>100</DefaultValue>
+    </Constraint>
     <ExtendedCapabilities>
       <wadl:application xmlns:wadl="http://wadl.dev.java.net/2009/02">
         <wadl:resources base="http://localhost:1999/">

